event-adapter/cmd/provider/provider.go

```diff
@@ -17,44 +17,146 @@ limitations under the License.
+		restClient:         restClient,
+		service:            stackdriverService,
+		maxRetrievedEvents: maxRetrievedEvents,
+		sinceMillis:        sinceMillis,
+	}
 }
 
 // GetNamespacedEventsByName gets the event with the given namespace and name
 func (p *StackdriverProvider) GetNamespacedEventsByName(namespace, eventName string) (*api.Event, error) {
-	return nil, fmt.Errorf("GetNamespacedEventsByName is not implemented yet.")
+
+	standardFilter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+	filter := "(" + standardFilter + " AND jsonPayload.metadata.namespace = \"" + namespace + "\" AND jsonPayload.metadata.name = \"" + eventName + "\")"
+	event, err := p.doRequest(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(event.Items) == 0 {
+		return nil, fmt.Errorf("Event not found")
+	}
+	return &event.Items[0], nil
 }
 
 // ListAllEventsByNamespace gets all the events with the given namespace
 func (p *StackdriverProvider) ListAllEventsByNamespace(namespace string) (*api.EventList, error) {
-	return nil, fmt.Errorf("ListAllEventsByNamespace is not implemented yet.")
+
+	standardFilter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+
+	filter := "(" + standardFilter + " AND jsonPayload.metadata.namespace = \"" + namespace + "\")"
+	if err != nil {
+		return nil, err
+	}
+
+	return p.doRequest(filter)
 }
 
 // ListAllEvents gets all the events
 func (p *StackdriverProvider) ListAllEvents() (*api.EventList, error) {
-	return nil, fmt.Errorf("ListAllEvents is not implemented yet.")
+	filter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+	return p.doRequest(filter)
+}
+
+func standardFilter() (string, error) {
+	projectID, err := gce.ProjectID()
+
+	if err != nil {
+		return "", fmt.Errorf("Cannot retrieve projectID. %v", err)
+	}
+
+	return "(logName = \"projects/" + projectID + "/logs/events\" AND jsonPayload.kind = \"Event\")", nil
+
 }
 
 // CreateNewEvent creates a new event in the given namespace
 func (p *StackdriverProvider) CreateNewEvent(namespace string) (*api.Event, error) {
 	return nil, fmt.Errorf("CreateNewEvent is not implemented yet.")
 }
+
+func (p *StackdriverProvider) doRequest(filter string) (*api.EventList, error) {
+	projectID, err := gce.ProjectID()
+
+	if err != nil {
+		return nil, fmt.Errorf("Cannot retrieve projectID. %v", err)
+	}
+
+	resource := []string{
+		"projects/" + projectID,
+	}
+
+	entriesService := sd.NewEntriesService(p.service)
+	translator := newEventsTranslator(p.maxRetrievedEvents)
+
+	var timeLimit time.Time
+	if p.sinceMillis < 0 {
+		timeLimit = time.Now().Add((-3600) * time.Second)
+	} else {
+		timeLimit = time.Unix(0, p.sinceMillis*1e6)
+	}
+	timeFilter := "(" + filter + " AND " + "jsonPayload.metadata.creationTimestamp >= \"" + timeLimit.Format("2006-01-02T15:04:02.002Z") + "\")"
+
+	eventsList := &api.EventList{}
+	var response *sd.ListLogEntriesResponse
+
+	for redirection := 0; redirection < 1000; redirection++ {
```

Comment:
Please don't use such literals (1000) in the code unless it's very clear what do they mean and where did the value come from. If it's not obvious why this value is chosen, it's better to comment the reasoning or create a constant or flag. It's easy to end up with such literals scattered all over the code, which may sometimes cause unexpected and hard to explain behavior (as this value isn't documented anywhere).

Reply:
Removed